vertx-service-discovery/src/main/java/io/vertx/servicediscovery/impl/DefaultServiceDiscoveryBackend.java

```diff
@@ -34,10 +36,14 @@
  */
 public class DefaultServiceDiscoveryBackend implements ServiceDiscoveryBackend {
   private AsyncMap<String, String> registry;
+  private VertxInternal vertx;
```

Comment:
Sorry if that was unclear. You don't need casting to `VertxInternal` since you have `isClustered` in `Vertx` and `getClusterWideMap` in `SharedData`